File: app/src/main/java/com/mytlogos/enterprise/background/room/model/RoomFailedEpisode.java
package com.mytlogos.enterprise.background.room.model;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(foreignKeys = @ForeignKey(
        entity = RoomEpisode.class,
        parentColumns = "episodeId",
        childColumns = "episodeId",
        onDelete = ForeignKey.CASCADE
))
public class RoomFailedEpisode {
    @PrimaryKey
    private final int episodeId;
    private final int failCount;

    public RoomFailedEpisode(int episodeId, int failCount) {
        this.episodeId = episodeId;
        this.failCount = failCount;
    }

    public int getEpisodeId() {
        return episodeId;
    }

    public int getFailCount() {
        return failCount;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        RoomFailedEpisode that = (RoomFailedEpisode) o;

        if (getEpisodeId() != that.getEpisodeId()) return false;
        return getFailCount() == that.getFailCount();
    }

    @Override
    public int hashCode() {
        int result = getEpisodeId();
        result = 31 * result + getFailCount();
        return result;
    }
}

File: app/src/main/java/com/mytlogos/enterprise/background/room/model/RoomFailedEpisode.kt
package com.mytlogos.enterprise.background.room.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    foreignKeys = [
        ForeignKey(
            entity = RoomEpisode::class,
            parentColumns = ["episodeId"],
            childColumns = ["episodeId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
class RoomFailedEpisode(@field:PrimaryKey val episodeId: Int, val failCount: Int) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as RoomFailedEpisode
        return episodeId == that.episodeId && failCount == that.failCount
    }

    override fun hashCode(): Int {
        var result = episodeId
        result = 31 * result + failCount
        return result
    }
}